src/H5Gdeprec.c

```diff
@@ -1133,7 +1127,7 @@ H5Gget_objname_by_idx(hid_t loc_id, hsize_t idx, char *name /*out*/, size_t size
     size_t               name_len = 0; /* Length of object name */
     ssize_t              ret_value;    /* Return value */
 
-    FUNC_ENTER_API(FAIL)
+    FUNC_ENTER_API(-1)
```

Comment:
Why replace the use of `FAIL` here?

Reply:
Because FAIL is semantically an herr_t value. -1 is better for an ssize_t, even though they are the same.